neighsyncd/neighsync.cpp

```diff
@@ -11,12 +11,244 @@
 #include "linkcache.h"
 
 #include "neighsync.h"
+#include "warm_restart.h"
 
 using namespace std;
 using namespace swss;
 
-NeighSync::NeighSync(DBConnector *db) :
-    m_neighTable(db, APP_NEIGH_TABLE_NAME)
+const NeighRestartAssit::cache_state_map NeighRestartAssit::cacheStateMap =
+{
+    {STALE,     "STALE"},
+    {SAME,      "SAME"},
+    {NEW,       "NEW"},
+    {DELETE,    "DELETE"},
+    {UNKNOWN,   "UNKNOWN"}
+};
+
+NeighRestartAssit::NeighRestartAssit(RedisPipeline *pipelineAppDB,
+  RedisPipeline *pipelineConfDB):
+    m_appTable(pipelineAppDB, APP_NEIGH_TABLE_NAME, false),
+    m_cfgTable(pipelineConfDB, CFG_WARM_RESTART_TABLE_NAME, false)
+{
+
+    m_appTableName = m_appTable.getTableName();
+
+    if (!WarmStart::isWarmStart())
+    {
+        m_warmStartInProgress = false;
+    }
+    else
+    {
+        uint32_t temp_value = WarmStart::getWarmStartTimer("neighsyncd");
+        if (temp_value != 0)
+        {
+            m_reconcileTimer = temp_value;
+        }
+    }
+}
+
+NeighRestartAssit::~NeighRestartAssit()
+{
+}
+
+/* join the field-value strings for straight printing */
+string NeighRestartAssit::joinVectorString(const vector<FieldValueTuple> &fv)
+{
+    string s;
+    for (const auto &temps : fv )
+    {
+	   s += temps.first + ":" + temps.second + ", ";
+    }
+    return s;
+}
+
+void NeighRestartAssit::setCacheEntryState(std::vector<FieldValueTuple> &fvVector,
+    cache_state_t state)
+{
+    fvVector.back().second = cacheStateMap.at(state);
+}
+
+NeighRestartAssit::cache_state_t NeighRestartAssit::getCacheEntryState(const std::vector<FieldValueTuple> &fvVector)
+{
+    for (auto &iter : cacheStateMap)
+    {
+        if (fvVector.back().second == iter.second)
+        {
+            return iter.first;
+        }
+    }
+    assert("Cache Entry State is corrupted" == NULL);
+    return UNKNOWN;
+}
+
+/* Read table from APPDB and append stale flag then insert to cachemap */
+void NeighRestartAssit::readTableToMap()
+{
+    vector<string> keys;
+
+    m_appTable.getKeys(keys);
+    FieldValueTuple state(CACHE_STATE_FIELD, "");
+
+    for (const auto &key: keys)
+    {
+        vector<FieldValueTuple> fv;
+
+	    /* if the fieldvlaue empty, skip */
+        if (!m_appTable.get(key, fv))
+        {
+            continue;
+        }
+
+        fv.push_back(state);
+        setCacheEntryState(fv, STALE);
```

Comment:
What is the reason it is split into two steps? 
it looks to me "fv.emplace_back(CACHE_STATE_FIELD, cacheStateMap.at(STALE))"  should work well?